lib/dep_selector/dependency_graph.rb

```diff
@@ -60,11 +60,17 @@ def generate_gecode_wrapper_constraints(packages_to_include_in_solve=nil)
             packages.map{ |name, pkg| pkg }
           end
 
+        logId = SecureRandom.uuid
         debugFlag = DebugOptionFile && File::exists?(DebugOptionFile)
+        Debug.log.level = Logger::INFO unless debugFlag
+        Debug.log.formatter = proc do |severity, datetime, progname, msg|
+          "#{logId}: #{msg}\n"
+        end
```

Comment:
Seems like all this could go up into `Debug` and you could have a method like `Debug.enable_verbose` to set the log level to debug.